src/garrysmod/gamemodes/zombiesurvival/gamemode/cl_scoreboard.lua

```diff
@@ -1,216 +1,249 @@
-	gui.EnableScreenClicker(false)
+    gui.EnableScreenClicker(false)
 
-	if ScoreBoard then
-		PlayMenuCloseSound()
-		ScoreBoard:SetVisible(false)
-	end
+    if ScoreBoard then
+        PlayMenuCloseSound()
+        ScoreBoard:SetVisible(false)
+    end
 end
 
 local PANEL = {}
 
 PANEL.RefreshTime = 2
 PANEL.NextRefresh = 0
-PANEL.m_MaximumScroll = 0
-
-local function BlurPaint(self)
-	draw.SimpleTextBlur(self:GetValue(), self.Font, 0, 0, self:GetTextColor())
-
-	return true
-end
 
 function PANEL:Init()
-	self.NextRefresh = RealTime() + 0.1
-
-	self.m_TitleLabel = vgui.Create("DLabel", self)
-	self.m_TitleLabel.Font = "ZSScoreBoardTitle"
-	self.m_TitleLabel:SetFont(self.m_TitleLabel.Font)
-	self.m_TitleLabel:SetText(GAMEMODE.Name)
-	self.m_TitleLabel:SetTextColor(COLOR_GRAY)
-	self.m_TitleLabel:SizeToContents()
-	self.m_TitleLabel:NoClipping(true)
-	self.m_TitleLabel.Paint = BlurPaint
-
-	self.m_ServerNameLabel = vgui.Create("DLabel", self)
-	self.m_ServerNameLabel.Font = "ZSScoreBoardSubTitle"
-	self.m_ServerNameLabel:SetFont(self.m_ServerNameLabel.Font)
-	self.m_ServerNameLabel:SetText(GetHostName())
-	self.m_ServerNameLabel:SetTextColor(COLOR_GRAY)
-	self.m_ServerNameLabel:SizeToContents()
-	self.m_ServerNameLabel:NoClipping(true)
-	self.m_ServerNameLabel.Paint = BlurPaint
-
-	self.m_AuthorLabel = EasyLabel(self, "by "..GAMEMODE.Author.." ("..GAMEMODE.Email..")", "ZSScoreBoardPing", COLOR_GRAY)
-	self.m_ContactLabel = EasyLabel(self, GAMEMODE.Website, "ZSScoreBoardPing", COLOR_GRAY)
-
-	self.m_HumanHeading = vgui.Create("DTeamHeading", self)
-	self.m_HumanHeading:SetTeam(TEAM_HUMAN)
-
-	self.m_ZombieHeading = vgui.Create("DTeamHeading", self)
-	self.m_ZombieHeading:SetTeam(TEAM_UNDEAD)
-
-	self.m_PointsLabel = EasyLabel(self, "Score", "ZSScoreBoardPlayer", COLOR_GRAY)
-	self.m_RemortCLabel = EasyLabel(self, "R.LVL", "ZSScoreBoardPlayer", COLOR_GRAY)
-
-	self.m_BrainsLabel = EasyLabel(self, "Brains", "ZSScoreBoardPlayer", COLOR_GRAY)
-	self.m_RemortCZLabel = EasyLabel(self, "R.LVL", "ZSScoreBoardPlayer", COLOR_GRAY)
-
-	self.ZombieList = vgui.Create("DScrollPanel", self)
-	self.ZombieList.Team = TEAM_UNDEAD
-
-	self.HumanList = vgui.Create("DScrollPanel", self)
-	self.HumanList.Team = TEAM_HUMAN
-
-	self:InvalidateLayout()
+    self.NextRefresh = RealTime() + 0.1
+
+    self.m_Header = vgui.Create("Panel", self)
+    self.m_Header:Dock(TOP)
+    self.m_Header:DockPadding(16, 0, 16, 0)
+
+    self.m_Header.Paint = function(self, w, h)
+        surface.SetDrawColor(5, 5, 5, 220)
+        PaintGenericFrame(self, 0, 0, w, h, 32)
+    end
+
+    self.m_TitleLabel = vgui.Create("DLabel", self.m_Header)
+    self.m_TitleLabel:SetFont("ZSScoreBoardSubTitle")
+    self.m_TitleLabel:SetText(string.format("%s | %s", GAMEMODE.Name, translate.Get(game.GetMap())))
+    self.m_TitleLabel:SetTextColor(COLOR_GRAY)
+    self.m_TitleLabel:Dock(LEFT)
+    self.m_TitleLabel:SetContentAlignment(5)
+    self.m_TitleLabel:SizeToContents()
```

Comment:
Here, we're creating a panel to use as a positioning container for the title label, and docking it to the top of the root panel. Before, this title label was created, and it's position was set to a static coordinate within the `Panel:PerformLayout` method.